src/main/java/onlinebookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,84 @@
+package onlinebookstore.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.constraints.Positive;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import onlinebookstore.dto.book.BookDtoWithoutCategoryIds;
+import onlinebookstore.dto.category.CategoryDto;
+import onlinebookstore.dto.category.CategoryRequestDto;
+import onlinebookstore.service.BookService;
+import onlinebookstore.service.CategoryService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Category manager",
+        description = "Endpoints for categories management")
+@Validated
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @GetMapping("/{id}")
+    @PreAuthorize("hasAnyRole('ROLE_USER', 'ROLE_ADMIN')")
+    @Operation(summary = "Get category", description = "Get category by id")
+    public CategoryDto getCategoryById(@PathVariable @Positive Long id) {
+        return categoryService.getById(id);
+    }
+
+    @GetMapping
+    @PreAuthorize("hasAnyRole('ROLE_USER', 'ROLE_ADMIN')")
+    @Operation(summary = "Get all categories", description = "Get list of all categories")
+    public List<CategoryDto> getAll(Pageable pageable) {
```

Comment:
consider using `@PageableDefault` in case client did not provide pagination params